src/compile/unit.ts

```diff
@@ -105,8 +112,15 @@ export class UnitModel extends Model {
   private _initAxis(encoding: Encoding, config: Config): Dict<Axis> {
     return [X, Y].reduce(function(_axis, channel) {
       // Position Axis
-      if (vlEncoding.has(encoding, channel)) {
-        const axisSpec = encoding[channel].axis;
+      if (vlEncoding.has(encoding, channel) ||
+          (channel === X && vlEncoding.has(encoding, X2)) ||
+          (channel === Y && vlEncoding.has(encoding, Y2))) {
+
+        let targetChannel = channel;
+        if (!vlEncoding.has(encoding, channel)) {
+          targetChannel = channel === X ? X2 : Y2;
+        }
+        const axisSpec = encoding[targetChannel].axis;
```

Comment:
We haven't really merge the axis properties between `x` and `x2` if both exist yet.  
I would say just add `//TODO` for now and create a follow-up issue.  
No need to implement it yet.